### ISSUE TITLE
 CVE-2023-28155

### ISSUE THREAD

**Issue by khitrenovich** [cypress-io/request#27]
Being a fork of the original `request` package, `@cypress/request` is vulnerable to [CVE-2023-28155](https://github.com/advisories/GHSA-p8p7-x288-28g6) (SSRF via insecure redirects). There is [an issue](https://github.com/request/request/pull/3442) and [an open PR](https://github.com/request/request/pull/3444) in the `request` repo, but the fix is unlikely to be merged since `request` is long abandoned. This fork is the second most popular fork, it would be great to have it patched.

**Comment by MikeMcC399** [comment]
For `@cypress/request@2.88.12` (currently the latest version)

`npm audit` on `cypress@12.17.3` and earlier versions reports:

```text
@cypress/request  *
Severity: moderate
Server-Side Request Forgery in Request - https://github.com/advisories/GHSA-p8p7-x288-28g6
Depends on vulnerable versions of tough-cookie
fix available via `npm audit fix --force`
Will install cypress@4.2.0, which is a breaking change
node_modules/@cypress/request
  cypress  >=4.3.0
  Depends on vulnerable versions of @cypress/request
  node_modules/cypress
```

## Reference

- [CVE-2023-28155](https://nvd.nist.gov/vuln/detail/CVE-2023-28155)

**Comment by MikeMcC399** [comment]
If / when this is resolved, then it would make sense to also update

https://github.com/cypress-io/cypress/blob/develop/cli/package.json#L23 currently showing

`"@cypress/request": "^2.88.11",`

because otherwise updating an installed instance of Cypress does not pull in the latest version of `@cypress/request`.

**Comment by MikeMcC399** [comment]
- I see that PR https://github.com/cypress-io/request/pull/28 has now been merged 👍🏻 and [@cypress/request@3.0.0](https://github.com/cypress-io/request/releases/tag/v3.0.0) has been released.

**Comment by ST-DDT** [comment]
Here is the PR that adds the new release to cypress itself:

- https://github.com/cypress-io/cypress/pull/27495

**Comment by MikeMcC399** [comment]
@khitrenovich 

- Could you please check? I believe that the merged PR #28 and the release of [@cypress/request@3.0.0](https://github.com/cypress-io/request/releases/tag/v3.0.0) resolves this issue and that therefore this issue could be closed.

**Comment by khitrenovich** [comment]
We stopped using `@cypress/request` since the issue was opened, so I won't be able to verify that. I hope one of the people who upvoted the issue (more than dozen as of today) could chime in and help.

**Comment by heloufir** [comment]
Hello @MikeMcC399 
I can confirm that there is no vulnerability alert when I run `npm audit fix` after I changed the version of `@cypress/request` to `3.0.0` in my projet.

For my case I am using `cypress` to do `e2e` testing, and I was having the npm vulnerability alert, but after I added the below code to my `package.json` file, the alert is gone! thanks a lot to dev team:

```json
"overrides": {
  "@cypress/request": "^3.0.0"
}
```

**Comment by OleksiiKachan** [comment]
if you are using yarn then instead of `overrides` use `resolutions`

```  
  "resolutions": {
    "@cypress/request": "^3.0.0"
  }
```

**Comment by MikeMcC399** [comment]
- PR https://github.com/github/advisory-database/pull/2609 updates the GitHub Advisory https://github.com/advisories/GHSA-p8p7-x288-28g6 to mark this issue as being resolved by `@cypress/request@3.0.0`.

**Comment by nbouvrette** [comment]
This will be fixed on Cypress when https://github.com/cypress-io/cypress/pull/27495 is merged.

**Comment by verheyenkoen** [comment]
> This will be fixed on Cypress when [cypress-io/cypress#27495](https://github.com/cypress-io/cypress/pull/27495) is merged.

Appears to be a breaking change according to #37 https://github.com/cypress-io/cypress/pull/27495#issuecomment-1673320655. Will have to wait for Cypress 13 to be fully fixed.

**Comment by MikeMcC399** [comment]
This issue is about the vulnerability in `@cypress/request` which was fixed in [@cypress/request@3.0.0](https://github.com/cypress-io/request/releases/tag/v3.0.0).

To confirm that no vulnerability is reported when using `@cypress/request@3.0.0`:

```bash
mkdir cy-request-test
cd cy-request-test
npm init -y
npm install @cypress/request
npm ls
npm audit
```

shows:

```bash
cy-request-test@1.0.0
└── @cypress/request@3.0.0

found 0 vulnerabilities
```

So from a formal point of view this issue could be closed. It may however be helpful to leave it open until the issue has been fixed in Cypress itself.

---

The vulnerability in Cypress 12.x was reported in
- issue https://github.com/cypress-io/cypress/issues/27535
and is planned for resolution through
- PR https://github.com/cypress-io/cypress/pull/27495

**Comment by MikeMcC399** [comment]
[Cypress 13.0.0 changelog](https://docs.cypress.io/guides/references/changelog#13-0-0) for the version released today, Aug 29, 2023, states:

**Dependency Updates:**

- Upgraded [`@cypress/request`](https://www.npmjs.com/package/@cypress/request) from `^2.88.11` to `^3.0.0` to address the [CVE-2023-28155](https://github.com/advisories/GHSA-p8p7-x288-28g6) security vulnerability. Addresses [#27535](https://github.com/cypress-io/cypress/issues/27535). Addressed in [#27495](https://github.com/cypress-io/cypress/pull/27495).


There should be no further barriers to closing this issue.
